solidity/contracts/CronosGravity.sol

```diff
@@ -77,10 +78,15 @@ contract CronosGravity is Gravity, AccessControl, Pausable, Ownable {
     */
     function migrateToken(
         address _tokenContract,
-        address _destination,
-        uint256 _amount
+        address _newGravityAddress,
+        uint256 _amount,
+        bool isCosmosToken
     ) public onlyOwner {
-        IERC20(_tokenContract).safeTransfer(_destination, _amount);
+        if (isCosmosToken) {
+            ICosmosToken(_tokenContract).setGravityContract(_newGravityAddress);
+        } else {
+            IERC20(_tokenContract).safeTransfer(_newGravityAddress, _amount);
```

Comment:
For this migration feature, there are some concerns could come from community I think..the owner privilege looks excessive to the funds safe. 
Because based on the code, owner can move specific ERC20 funds to any specified `_newGravityAddress` address(although we won't do so).
Some optional recommendations for you:
- adding a `TimeLockedMigration` feature to notify/show users that a migration is initiated so far, the execution could be delayed for a specified time.  
- Dao governance to initiate migration.
- adding a contract status like `expiredFlag` which is timelock-controlled by owner, then owner can migrate tokens to public address at any time.

Reply:
added